src/client/menu/menu.c

```diff
@@ -2915,9 +2915,37 @@ Mods_MenuInit(void)
     int currentmod;
     int x = 0;
     int y = 0;
+	char modname[MAX_QPATH]; //TG626
 
     Mods_NamesInit();
-
+	
+	// create array of bracketed display names from folder names - TG626
+	char **displaynames=malloc(sizeof(*displaynames) * nummods);
+	
+	for (int i=0; i < nummods; i++)
+	{
+		strcpy(modname, "[");
+		if (strlen(modnames[i]) < 16)
+		{
+			strcat(modname, modnames[i]);
+			for (int j=0; j < 15 - strlen(modnames[i]); j++)
+			{
+				strcat(modname, " ");
+			}
+		}
+		else
+		{
+			strncat(modname, modnames[i], 12);
```

Comment:
strncat() is dangerous - it doesn't guarantee \0-termination!
So after the char from `modnames[i][11] it won't insert a \0 character, and the call to strcat() below might start there.

it's possible that all this would be easier with snprintf() with the correct arguments (you can specify how many chars of a '%s' argument should be copied).

Reply:
It doesn't cause any crashes - short names, 60 character long names were both tested, tho I see your point.